src/analyze.ts

```diff
@@ -876,6 +884,14 @@ export default async function analyze(
                 pjsonPath = path.resolve(pjsonPath, '../../package.json');
               if (pjsonPath !== rootPjson) assets.add(pjsonPath);
               break;
+            case MODULE_FN:
+              if (
+                node.arguments.length &&
+                node.arguments[0].type === 'Literal'
+              ) {
+                await processRequireArg(node.arguments[0]);
```

Comment:
It looks like `module.register(specifier[, parentURL][, options])` has more than one argument.

Do we need to handle these as well?

Reply:
Also it should be handled as an import, not require:

> Customization hooks to be registered; this should be the same string that would be passed to import(), except that if it is relative, it is resolved relative to parentURL.

https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options

Something like this:

https://github.com/timfish/nft/blob/db7751dc417f0de81d5b76141ea2e392deb3ecd2/src/analyze.ts#L647-L648

Reply:
`parentURL` is usually always `import.meta.url` in ESM. Do we need to cater for other `parentURL`s, or can we safely assume its relative to the current file?

Reply:
It could be any URL, correct? 

A quick search on google shows its not always `import.meta.url`


https://github.com/search?type=code&q=%22+module.register%28%22+language%3ATypeScript

https://github.com/dai-shi/waku/blob/aa8b021a2f1ab073abc601129b5d0d6c3bda8ca1/packages/waku/src/lib/renderers/dev-worker-impl.ts#L38

Reply:
`parentURL` could potentially be any URL but `import.meta.url` and `pathToFileURL('./')` both essentially mean:

> if `specifier` is a relative path, resolve relative to the current file/directory

Code could potentially pass a different URL here but I suspect that is not common. When I'm back online after this flight I'll have a more thorough search!

Reply:
I think we need to be careful here because we might trace the wrong file if we don't consider that parent url.

I believe we support `require.resolve()` and that can be relative to a path, as well as `await import()` although not sure if that one can be relative to a path.

Reply:
It's probably enough to only resolve  `module.register` when we can statically analyse the parentURL trivially and only the simple/common cases matter. We can skip resolving everything else and log a warning or something.

`require.resolve()` does not behave exactly the same as the default ESM resolver so we should not be using that for resolving ESM. And as far as I know, node doesn't expose its default ESM resolver either which makes this painful. 

For example, `require.resolve()` automatically tries multiple file extensions but the ESM resolver expects file extensions to be included.

Reply:
Actually there's `import.meta.resolve()` from node v20.6.0 but I think we can only access that from ESM 😂

Reply:
I think its fine if we don't support all possible cases - meaning if we can't statically analyze then we don't emit.

The thing I'm not okay with is emitting the wrong files because this is recursive so walking down the wrong tree can lead to severe bloat.

So perhaps starting with these two common cases:

- `import.meta.url`
- `pathToFileURL('./')`

Then add a TODO comment mentioning that more complex values are not handled at this time.